include/util/threadsafe_stat_phases.hpp

```diff
@@ -0,0 +1,170 @@
+/*******************************************************************************
+ * include/util/threadsafe_stat_phases.hpp
+ *
+ * Copyright (C) 2018 Marvin Löbel <loebel.marvin@gmail.com>
+ *
+ * All rights reserved. Published under the BSD-2 license in the LICENSE file.
+ ******************************************************************************/
+
+#include <chrono>
+#include <memory>
+#include <mutex>
+#include <sstream>
+#include <thread>
+#include <type_traits>
+#include <fstream>
+
+struct chrono_clock_type {
+  /** @brief Best available clock. */
+  using clock_type =
+      typename std::conditional<std::chrono::high_resolution_clock::is_steady,
+                                std::chrono::high_resolution_clock,
+                                std::chrono::steady_clock>::type;
+
+  using time_point = clock_type::time_point;
+  inline static time_point now() {
+    // static std::mutex m;
+    // m.lock();
+    auto r = clock_type::now();
+    // m.unlock();
+    return r;
+  }
+  inline static double milli_duration(time_point from, time_point to) {
+    using Dur =
+        std::chrono::duration<double, std::chrono::milliseconds::period>;
+    return Dur(to - from).count();
+  }
+};
+
+struct openmp_clock_type {
+  using time_point = double;
+  inline static time_point now() {
+    return omp_get_wtime();
+  }
+  inline static double milli_duration(time_point from, time_point to) {
+    return (to - from) * 1000.0;
+  }
+};
+
+using clock_type = chrono_clock_type;
+
+struct TsStatPhase {
```

Comment:
1. We use snake_case instead of camelCase (throughout the whole file)
2. Is this used somewhere?

Reply:
This file is currently no used, but I wanted to keep it in case I need to quickly debug some multithreading issues again.

I can keep the file somewhere else outside the repo though if you want.

Reply:
Nevermind, I just put it into its own branch.